### PR TITLE
Pdf backend

### PR DESCRIPTION
**Pull request by Tuan333**
## PR Summary
Small pull request to address #8142. The issue seems to propagate from a feature enhancement #5596 where `Line2D.get_markeredgecolor()` returns `None` due to the change-set. The fix simply check if `gc._rgb == None` in backendpdf (caused the said change-set), if so, set `gc._rgb` to be the value it would have taken if the change-set was not there (pre #5596 behavior). Hence, this can probably be viewed as an addition to #5596 where behaviour of backendpdf remains the same .

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_pdf.py [hunk 1]
```diff
@@ -1625,6 +1625,8 @@ def check_gc(self, gc, fillcolor=None):
         gc._fillcolor = fillcolor
 
         orig_alphas = getattr(gc, '_effective_alphas', (1.0, 1.0))
+        if gc.get_rgb() is None:
+            gc.set_foreground([0.0, 0.0, 0.0, 1.0])
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_pdf.py]: Why black?  This should probably have 0 alpha?

**Comment by tacaswell** [comment]
Can you add a test?  How to the png and pdf look with this change?

File: lib/matplotlib/backends/backend_pdf.py (hunk 1 above)
Comment (Tuan333) [inline lib/matplotlib/backends/backend_pdf.py]: I'm not entirely sure what the alpha value has to do. I simply revert the unwanted behavior of #5596 i.e. in the function Line2D.draw(renderer), with the above setting, prior to the the change-set, at line `830`, `edgecolor` would be `'k'`, and after the call `gc.set_foreground(edgecolor)` on line `835`, `gc._rgb` takes the value that I set. Will put a test case tomorrow to check to be sure that I did not miss anything.

**Comment by Tuan333** [comment]
Note: I'm having a bit of difficulty comparing PDF image to png image, so I might modify your compare_image function and include it in the PR to make this process easier later too.

**Comment by tacaswell** [comment]
Sorry, I meant to just look at them, not to progamtically compare them!

**Comment by Tuan333** [comment]
Ah, in that case, it should be an easy enough job. Been hacking through compare_image function and ghost script in the past few days in vain. Will get back to this one.

**Pull request closed without merging**